templates/Web/Pages/React.List/src/components/ReactList/ListForm.jsx

```diff
@@ -1,20 +1,41 @@
-﻿import React from "react";
+﻿import React, {Component} from 'react';
 
-export default function ListForm(props) {
-  const { onAddListItem, textField, onChangeInputText } = props;
-  return (
-    <form onSubmit={onAddListItem} className="input-group my-3">
-      <input
-        type="text"
-        onChange={event => onChangeInputText(event, "textField")}
-        value={textField}
-        className="form-control"
-        placeholder="Add text here..."
-        aria-label="Add text here..."
-      />
-      <button type="submit" className="btn btn-primary ml-2">
-        Submit
-      </button>
-    </form>
-  );
-}
+export default class ListForm extends Component {
```

Comment:
quick question why did you do the move from a functional component to a class component?

Reply:
If it was because of the state initialization of the text field isn't this actually a good opportunity to start migrating slowly towards react hooks? 

Instead we could do something like this
```js
const [textField, setTextField] = useState("");
```

textField would be the readonly value of textField, while setTextField() would set the value of the said text field.

Reply:
Any thoughts @piotrmark?

Reply:
React Hooks is kind of a [controversial](https://news.ycombinator.com/item?id=18302162) addition to React. We would have to think about it more from the perspective of shipping (opinionated) best-practice templates.

Reply:
All good just wanted to have some input on if we should or not.

Reply:
Yeah, React hooks is a super fresh feature (maybe it's not experimental any more, but probably not well established in community yet).
If we want to move to hooks, IMO it would need to be one big switch instead of changing one component at a time as probably we don't want inconsistent templates.
Regardless, it's out of scope for this task which is about eliminating the warning about input management.

Reply:
It is definitely super fresh came out in 16.8 I think. We are using it on our client side right now!

Yeah I realized that after the fact that it would be a better choice if we did a full overhaul with hooks if we did choose to do it. You are completely correct about the inconsistencies created with not doing such. 

Yup definitely out of scope just the switch was a good time to bring up the question.